### ISSUE TITLE
No <asm/byteorder.h> on Solaris SPARC

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#479]
**Reported by darkAudax on 6 Aug 2008 12:20 UTC**

source: http://tinyshell.be/aircrackng/forum/index.php?topic=3932.0

If one tries to compile the latest SVN sources (build 1128) on a Sun workstation running Solaris 10, it fails. The first reason I reported (unportable test in evalrev). That affects Solaris on both SPARC and x86 platforms. Another is the attempted inclusion of the header file <asm/byteorder.h> on Solaris SPARC. (This is not an issue on Solaris x86).

The following in aircrack-ng/src/osdep/network.c

```
#ifdef __MACH__
#include <libkern/OSByteOrder.h>
#else
#include <asm/byteorder.h>
#endif /* __MACH__ */
```

will try to include <asm/byteorder.h>. If this is modified to:

```
#ifdef __MACH__
#include <libkern/OSByteOrder.h>
#elif defined (__sun) && defined (__sparc) /* Solaris SPARC, not Solaris x86 */
#include <sys/byteorder.h>
#else
#include <asm/byteorder.h>
#endif /* __MACH__ */
```

then on Solaris, with the SPARC processor, this will include the right file.

Any chance of someone updating the source? It's only a couple of lines and should have no effect on any other platform.

Given Solaris x86 runs on most PC hardware (I'm typing this on a Sony laptop running Solaris x86), I think it would be good if Solaris x86 was supported. Given it is open-source, it should be relatively easy. Solaris on SPARC is perhaps less useful, but if Solaris x86 was supported, the SPARC would almost certainly work if the correct wireless card was fitted.

**Comment by aircrack-ng** [comment]
**Modified by misterx on 6 Aug 2008 12:20 UTC**

**Comment by aircrack-ng** [comment]
**Modified by misterx on 6 Aug 2008 12:20 UTC**

**Comment by aircrack-ng** [comment]
**Comment by misterx on 6 Aug 2008 12:20 UTC**

(In #1152) Fix compilation on Sun SPARC (Closes: #479).
